source/lac/scalapack.cc

```diff
@@ -472,6 +472,7 @@ template <typename NumberType>
 void ScaLAPACKMatrix<NumberType>::copy_transposed(const ScaLAPACKMatrix<NumberType> &B)
 {
   add(B,0,1,true);
+  state = B.state;
```

Comment:
i would say move that before `add()` and inside `add()` check that `state==B.state`?

Reply:
I definitely agree with moving the reassignment.
Matrix B could have, in my opinion, any state and therefore a check is not needed.
The user should be aware which operations are performed.